source/common/router/config_impl.h

```diff
@@ -321,6 +321,11 @@ class VirtualHostImpl : public VirtualHost, Logger::Loggable<Logger::Id::router>
                              scope.scopeFromStatName(stat_names.other_), stat_names) {}
   };
 
+  RouteConstSharedPtr
+  getRouteFromRoutes(const RouteCallback& cb, const Http::RequestHeaderMap& headers,
+                     const StreamInfo::StreamInfo& stream_info, uint64_t random_value,
+                     std::vector<RouteEntryImplBaseConstSharedPtr> routes) const;
```

Comment:
So, I will always copy the whole route list? May be `Span` would be better here to avoid any unnecessary stack allocation.

Reply:
Done. Looks like `Span` can also avoid copying, so changing vector to `Span` is enough, right?